.github/workflows/ci.yml

```diff
@@ -0,0 +1,197 @@
+        id: version_v_github
+        with:
+          owner: MichalLytek
+          repo: type-graphql
+          excludes: prerelease, draft
+
+      - name: Remove leading v* from GitHub version
+        id: version_github
+        run: |
+          _version="${{ steps.version_v_github.outputs.release }}"
+          printf 'value=%s\n' "${_version#?}" >> "$GITHUB_OUTPUT"
+
+      - name: Read Git tag version
+        id: version_gittag
+        run: |
+          _version="${{ github.ref_name }}"
+          printf 'value=%s\n' "${_version#?}" >> "$GITHUB_OUTPUT"
+
+      - name: Compare package.json with Git tag
+        uses: madhead/semver-utils@latest
+        id: comparison_package_gittag
+        with:
+          version: ${{ steps.version_package.outputs.value }}
+          compare-to: ${{ steps.version_gittag.outputs.value }}
+          lenient: false
+
+      - name: Compare Git tag with GitHub
+        uses: madhead/semver-utils@latest
+        id: comparison_gittag_github
+        with:
+          version: ${{ steps.version_gittag.outputs.value }}
+          compare-to: ${{ steps.version_github.outputs.value }}
+          lenient: false
+
+      - name: Check package.json == Git tag
+        run: |
+          if [ ! "${{ steps.comparison_package_gittag.comparison-result }}" = "=" ]; then
+            printf '[ERROR]: package.json (%s) != Git tag (%s)\n' "${{ steps.version_package.outputs.value }}" "${{ steps.version_gittag.outputs.value }}"
+            exit 1
+          fi
+
+      - name: Check Git tag > GitHub
+        run: |
+          if [ ! "${{ steps.comparison_gittag_github.comparison-result }}" = ">" ]; then
+            printf '[ERROR]: Git tag (%s) !> GitHub (%s)\n' "${{ steps.version_gittag.outputs.value }}" "${{ steps.version_github.outputs.value }}"
+            exit 1
+          fi
+
+  check:
+    name: Build & Lint & Tests
+    needs: version
+    if: always() && (needs.version.result == 'success' || needs.version.result == 'skipped')
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+      matrix:
+        node-version: [14.x, 16.x, 18.x]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Setup Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      - name: Install Dependencies
+        run: |
+          npm ci
+
+      - name: Build
+        run: |
+          npm run build
+
+      - name: Check
+        run: |
+          npm run check
+
+      - name: Test
+        run: npm run test:ci
+        env:
+          CI: true
+
+      - name: Upload code coverage
+        uses: codecov/codecov-action@v3
+        if: matrix.node-version == '18.x'
+
+  release:
+    name: Release package on NPM
+    needs: check
+    if: needs.check.result == 'success' && startsWith(github.ref, 'refs/tags/v')
```

Comment:
Could we make this npm version check and npm release a separate workflow?
Or it is intended to be a part of the build/lint/tests workflow to make sure we do not publish faulty version?